experimental/traffic-portal/src/app/core/profiles/profile-table/profile-table.component.ts

```diff
@@ -0,0 +1,148 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { Component, OnInit } from "@angular/core";
+import { UntypedFormControl } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute, Params } from "@angular/router";
+import { BehaviorSubject } from "rxjs";
+import { ResponseProfile } from "trafficops-types";
+
+import { ProfileService } from "src/app/api";
+import { CurrentUserService } from "src/app/shared/current-user/current-user.service";
+import { DecisionDialogComponent } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { ContextMenuActionEvent, ContextMenuItem } from "src/app/shared/generic-table/generic-table.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * ProfileTableComponent is the controller for the profiles page - which
+ * principally contains a table.
+ */
+@Component({
+	selector: "tp-profile-table",
+	styleUrls: ["./profile-table.component.scss"],
+	templateUrl: "./profile-table.component.html"
+})
+export class ProfileTableComponent implements OnInit {
+	/** All the physical locations which should appear in the table. */
+	public profiles: Promise<Array<ResponseProfile>>;
+
+  	/** Definitions of the table's columns according to the ag-grid API */
+	public columnDefs = [{
+		field: "cdnName",
+		headerName: "CDN"
+	}, {
+		field: "description",
+		headerName: "Description",
+	}, {
+		field: "id",
+		headerName: "ID",
+		hide: true
+	}, {
+		field: "lastUpdated",
+		headerName: "Last Updated",
+		hide: true
+	}, {
+		field: "name",
+		headerName: "Name"
+	}, {
+		field: "routingDisabled",
+		headerName: "Routing Disabled"
+	}, {
+		field: "type",
+		headerName: "Type"
+	}];
+
+	/** Definitions for the context menu items (which act on augmented cache-group data). */
+	public contextMenuItems: Array<ContextMenuItem<ResponseProfile>> = [
+		{
+			action: "delete",
+			multiRow: false,
+			name: "Delete"
+		},
+		{
+			href: "/core/servers",
+			name: "View Servers",
+			queryParams: (profile: ResponseProfile): Params => ({profileName: profile.name})
+		}
+	];
+
+	/** A subject that child components can subscribe to for access to the fuzzy search query text */
+	public fuzzySubject: BehaviorSubject<string>;
+
+	/** Form controller for the user search input. */
+	public fuzzControl: UntypedFormControl = new UntypedFormControl("");
```

Comment:
This should be a `FormControl<string>` with `nonNullable` set to true (you can see an example on the coordinates-table), these were converted to `UntypedFormControl`s during an angular upgrade and we should ideally be changing these to the typed version when we can.

Reply:
thanks for the example, Done.